Library/Formula/berkeley-db.rb

```diff
@@ -33,6 +33,20 @@ def install
       --enable-cxx
       --enable-compat185
     ]
+
+    if build.with? "java"
+      java_home = ENV["JAVA_HOME"] = `/usr/libexec/java_home`.chomp
+
+      # The Oracle JDK puts jni.h into #{java_home}/include and jni_md.h into
+      # #{java_home}/include/darwin.  The original Apple/SUN JDK placed jni.h
+      # and jni_md.h into
+      # /System/Library/Frameworks/JavaVM.framework/Versions/Current/Headers/
+      #
+      # Setup the include path with the Oracle JDK location first and the Apple JDK location second.
+      ENV["CFLAGS"] = "-I#{java_home}/include" <<
+                      " -I#{java_home}/include/darwin" <<
+                      " -I/System/Library/Frameworks/JavaVM.framework/Versions/Current/Headers/"
+    end
```

Comment:
@mikemcquaid I suggest we put these code into `JavaDependency`. My idea is to add new tag argument `:jdk` such as `depends_on :java => ["1.8", :jdk]`. If the `:jdk` is passed, it will check the `javac` and `jni.h` existence. At the same time, it will also set the environment variables `JAVA_HOME` and `JAVA_INCLUDE`, prepend `ENV["JAVA_HOME"]/"bin"` into `PATH`.

Therefore, it can be helpful for multiple formulae. I will try to create a PR, but I wonder how to set the superenv inside the  `Requirement`?